### DISCUSSION TITLE
Shouldn't every command support output streaming?

### DISCUSSION THREAD

**Discussion by inequity** [todddeluca/python-vagrant#110] [General]
I know at least 'provision' would be nearly as valuable to stream as 'up', but also things like long-running SSH commands would also be really nice to get a stream from. 

I may look into adding this myself and sending you a PR, just wondering what your thoughts were on it. Any specific reason that only 'up' supports this output mechanism right now?

**Comment by todddeluca** [comment]
Well, technically both `up` and `reload` support output streaming, thanks to PR#56, but I see what you mean: any other command that invokes `_call_vagrant_command` could theoretically support `_stream_vagrant_command`.  If you would like to submit a PR to add streaming to other commands, like `provision`, I would be happy to merge it.  

Adding streaming does change the return type of the command and add complexity to the code, so it might not be worth adding it to every compatible command until there is a need.  Anyway, I'm open to the idea.

It is also possible to stream output to a filehandle, using the `out_cm` parameter of the constructor, which works for some people's needs.

Cheers,
Todd
